File: __iOS_BACKUP__/ShapeByte/features/workout/item/domain/DefaultItemSetHandler.swift
//
//  DefaultItemSetHandler.swift
//  ShapeByte
//
//  Created by Lang, Stefan [Shape Byte Tech] on 29.07.24.
//

import Foundation
import Combine

// TODO: implement correctly by using Repetative
class DefaultItemSetHandler: ItemSetHandling {

    fileprivate(set) var statePublisher: PassthroughSubject<ItemSet.State, Never>
        = PassthroughSubject<ItemSet.State, Never>()

    func start(set: ItemSet) { }

    // TODO: implement
    func pause() {

    }

}

File: shared/src/commonMain/kotlin/de/stefan/lang/shapebyte/features/workout/item/domain/DefaultItemSetHandler.kt
package de.stefan.lang.shapebyte.features.workout.item.domain

import de.stefan.lang.shapebyte.features.workout.item.data.ItemSet
import de.stefan.lang.shapebyte.features.workout.item.data.ItemSetState
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.flow.StateFlow

// TODO: Implement correctly
class DefaultItemSetHandler : ItemSetHandling {
    override val stateFlow: StateFlow<ItemSetState>
        get() = TODO("Not yet implemented")

    override fun start(set: ItemSet, scope: CoroutineScope) {
        TODO("Not yet implemented")
    }

    override fun pause() {
        TODO("Not yet implemented")
    }

    override fun resume(resumeScope: CoroutineScope) {
        TODO("Not yet implemented")
    }
}
